crates/ruff_python_semantic/src/model.rs

```diff
@@ -601,7 +601,8 @@ impl<'a> SemanticModel<'a> {
                     // `from os.path import join`          -> `join`
                     // `from os.path import join as join2` -> `join2`
                     BindingKind::FromImport(FromImport { qualified_name }) => {
-                        if let Some((target_module, target_member)) = qualified_name.split_once('.')
+                        if let Some((target_module, target_member)) =
+                            qualified_name.rsplit_once('.')
```

Comment:
With this code (`test.py`)
```python
import typing
from collections.abc import Sequence

def f(x: typing.Sequence) -> None:
    ...
```

Running `cargo run --bin ruff -- --no-cache --target-version py311 --select UP test.py`

Without this change:
```
error: Failed to create fix for NonPEP585Annotation: Unable to insert `Sequence` into scope due to name conflict
test.py:4:10: UP006 Use `collections.abc.Sequence` instead of `typing.Sequence` for type annotation
Found 1 error.
```

With this change:
```python
test.py:4:10: UP006 [*] Use `collections.abc.Sequence` instead of `typing.Sequence` for type annotation
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

(and fix is valid)

Reply:
This seems to now have been fixed by this PR already: https://github.com/astral-sh/ruff/pull/6102